src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.ConvertHelpers.cs

```diff
@@ -31,88 +31,133 @@ public static partial class TensorPrimitives
                 return true;
             }
 
-            if (IsInt32Like<TFrom>() && typeof(TTo) == typeof(float))
+            if (typeof(TFrom) == typeof(byte))
```

Comment:
The changes here:
- Reorder the  blocks by TFrom to be a more consistent ordering
- Restructure so that there's a consistent `if (TFrom) { if (TTo) { ... } }` structure, such that there's one outer if for a from type and then 1 or more inner blocks for each target to type
- Adds handling of byte to float
- Adds some additional cases we could support with the existing code, e.g. we already handled converting ushort to uint but we can also make that "just work" with ushort to int